### PR TITLE
travis: run in container infra & remove pypy hacks

### PR DESCRIPTION
**Pull request by theanalyst**
Using travis' container based infra which is faster & allows caching pip
and is faster, also after the environment update hy seems happy with
pypy2.5 so removing the hacks needed to pass that.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,8 +6,7 @@ python:
   - "2.7"
   - "3.3"
   - "3.4"
-cache:
-  - $HOME/.pip-cache
+cache: pip
```
Comment (berkerpeksag) [inline .travis.yml]: What  does "pip" mean here? An executable or a directory?

Comment (algernon) [inline .travis.yml]: http://docs.travis-ci.com/user/caching/#pip-cache

Comment (berkerpeksag) [inline .travis.yml]: Great! Thanks @algernon.

**Comment by berkerpeksag** [comment]
LGTM.

**Comment by berkerpeksag** [comment]
I'm going to merge this since mostly @theanalyst and I are touched the travis/tox/Makefile files and the changes look good to me.

**Pull request merged**